drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,1727 @@
+ *
+ * @param dev device struct
+ * @param reg_type value from enum gpio_pca_series_reg_type
+ * @param buf pointer to new data to update from.
+ * @return int 0        if success
+ *             -EINVAL  if invalid arguments
+ *             -EACCES  if register is uncacheable
+ */
+static inline int gpio_pca_series_reg_cache_update(const struct device *dev,
+						   enum gpio_pca_series_reg_type reg_type,
+						   const uint8_t *buf)
+{
+	struct gpio_pca_series_data *data = dev->data;
+	uint8_t offset = gpio_pca_series_reg_cache_offset(dev, reg_type);
+	uint32_t size = gpio_pca_series_reg_size(dev, reg_type);
+	uint8_t *dst;
+
+#ifdef GPIO_NXP_PCA_SERIES_DEBUG
+	if (!buf) {
+		return -EINVAL;
+	}
+
+	if (offset == PCA_REG_INVALID) {
+		LOG_ERR("can not update non-cacheable reg type %d", reg_type);
+		return -EACCES;
+	}
+#endif /* GPIO_NXP_PCA_SERIES_DEBUG */
+
+	LOG_DBG("cache update type %d len %d from %s", reg_type, size,
+		(buf ? "buffer" : "device"));
+
+	dst = ((uint8_t *)data->cache) + offset;
+	LOG_DBG("cache write mem addr 0x%8.8lx len %d", (size_t)dst, size);
+
+	/** update cache from buf */
+	memcpy(dst, buf, size);
+
+	return 0;
+}
+
+#else /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+#define gpio_pca_series_reg_cache_read gpio_pca_series_reg_read
+
+struct gpio_pca_series_reg_cache_mini {
+	uint32_t output;   /** cache output value for faster output */
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	uint32_t input_old;    /** only used when interrupt mask & edge config is not present */
+	uint32_t int_rise; /** only used if interrupt edge is software-compared */
+	uint32_t int_fall; /** only used if interrupt edge is software-compared */
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+};
+
+static inline struct gpio_pca_series_reg_cache_mini *gpio_pca_series_reg_cache_mini_get(
+	const struct device *dev)
+{
+	struct gpio_pca_series_data *data = dev->data;
+	struct gpio_pca_series_reg_cache_mini *cache =
+		(struct gpio_pca_series_reg_cache_mini *)(&data->cache);
+	LOG_DBG("mini cache addr 0x%8.8lx", (size_t)cache);
+	return cache;
+}
+
+static inline int gpio_pca_series_reg_cache_mini_reset(const struct device *dev)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_reg_cache_mini *cache =
+		gpio_pca_series_reg_cache_mini_get(dev);
+	int ret = 0;
+
+	ret = gpio_pca_series_reg_read(dev, PCA_REG_TYPE_1B_OUTPUT_PORT,
+		(uint8_t *)&cache->output);
+	if (ret) {
+		LOG_ERR("minimum cache failed to read initial output: %d", ret);
+		goto out;
+	}
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	cache->int_rise = 0;
+	cache->int_fall = 0;
+
+	/* Read initial input value */
+	enum gpio_pca_series_reg_type input_reg =
+		cfg->part_cfg->flags & PCA_HAS_OUT_CONFIG ?
+		PCA_REG_TYPE_1B_INPUT_STATUS : PCA_REG_TYPE_1B_INPUT_PORT;
+
+	ret = gpio_pca_series_reg_read(dev, input_reg, (uint8_t *)&cache->input_old);
+	if (ret) {
+		LOG_ERR("minimum cache failed to read initial input: %d", ret);
+	}
+#else
+	UNUSED(cfg);
```

Comment:
I believe we should be able to replace these calls to `UNUSED` with `ARG_UNUSED`, and remove the `UNUSED` macro

Reply:
yes.